src/rgw/rgw_data_sync.cc

```diff
@@ -5119,7 +5090,11 @@ int RGWSyncBucketShardCR::operate(const DoutPrefixProvider *dpp)
       tn->log(5, SSTR("incremental sync on bucket failed, retcode=" << retcode));
       return set_cr_error(retcode);
     }
-    // TODO: handle transition to StateStopped
+
+    if (sync_status.state == rgw_bucket_shard_sync_info::StateStopped) {
+      bucket_state = BucketSyncState::Stopped;
```

Comment:
can we change this `BucketSyncState& bucket_state` to a `bool& bucket_sync_stopped`? we shouldn't overwrite the value of `RGWSyncBucketCR::bucket_state.state`, so that its value always matches what we last read-from/wrote-to rados

when we return back to `RGWSyncBucketCR`, we'll want to re-read `info.syncstopped` to decide between `Stopped` and `Init`. if do we find `info.syncstopped == true`, we'll use the value of `bucket_state.state` to decide whether or not we need to write `Stopped` back to rados. but if we overwrite `bucket_state = BucketSyncState::Stopped` here, we'd skip that write and the status object would still hold `state=Incremental`

Reply:
but before writing, i read the status from the file (to do the read-modify-write). 
so, I'm not sure why this should matter?